modules/node-shared/src/main/scala/org/tessellation/node/shared/domain/event/PeerIdEventCutter.scala

```diff
@@ -0,0 +1,9 @@
+package org.tessellation.node.shared.domain.event
+
+import org.tessellation.schema.gossip
+import org.tessellation.schema.peer.PeerId
+
+trait PeerIdEventCutter[F[_], Event] {
+  type PeerIdEvents = Map[PeerId, List[(gossip.Ordinal, Event)]]
```

Comment:
`import org.tessellation.schema.gossip.Ordinal`. should it stay an `Ordinal` or should it be `SnapshotOrdinal`?

Reply:
@marcinwadon The data structure that I need to work with is returning `gossip.Ordinal` which is not the same as `SnapshotOrdinal`. I can change it import the class directly if you like. I left it like that so that it was obvious that it wasn't the same type of ordinal.

Reply:
afaik we always use direct imports.
why you need to rely on the gossip ordering?

Reply:
my question was why the gossip is in here. how does it connect with the gossip?

Reply:
Put another way, I need to act on the result of this call:
```
ConsensusStorage.pullEvents(upperBound: Bound): F[Map[PeerId, List[(Ordinal, Event)]]]
```
and the only type variable is `Event`, which is either a `GlobalSnapshotEvent` or `CurrencySnapshotEvent` -- the others are fixed.

Reply:
@marcinwadon I could define the trait this way,
```
trait EventCutter[F[_], K, T, Event] {
  def cut(events: Map[K, List[(T, Event)]]): F[Map[K, List[(T, Event)]]]
}
```
and in the _make_ functions where it's passed around I could hard-code _K_ = `PeerId` and _T_ = `Ordinal`.
Would that be better?

Reply:
Ok, got it. So, the answer for my question is that the gossip ordinal is used to determine the order of events. That's fine to keep the gossip ordinal hardcoded then.

Reply:
Done